go/vt/vtgate/resilient_srv_topo_server.go

```diff
@@ -299,47 +328,54 @@ func (server *ResilientSrvTopoServer) GetSrvKeyspace(ctx context.Context, cell,
 	entry.mutex.Lock()
 	defer entry.mutex.Unlock()
 
-	// If the entry exists, return it
-	if entry.value != nil {
-		return entry.value, nil
+	// If the watch is already running, return the value
+	if entry.watchRunning {
+		return entry.value, entry.lastError
 	}
 
-	// not in cache, get the real value
+	// watch is not running, let's try to start it
 	newCtx := context.Background()
-
-	// start watching
-	notifications, stopWatching, err := server.topoServer.WatchSrvKeyspace(newCtx, cell, keyspace)
+	notifications, err := server.topoServer.WatchSrvKeyspace(newCtx, cell, keyspace)
```

Comment:
Do we want to keep newCtx here? That has been initialized to context.Background() before and would be unbounded.

Do we want to keep it that way or pass through "ctx" instead?

If we keep newCtx, can you please add a comment at its declaration why we use it instead?